Source/WebCore/Modules/webcodecs/WebCodecsEncodedVideoChunk.idl

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// FIXME: Support Serializable and Transferable.
+[
+    Conditional=WEB_CODECS,
+    EnabledBySetting=WebCodecsEnabled,
+    Exposed=(Window,DedicatedWorker),
+    InterfaceName=EncodedVideoChunk,
+] interface WebCodecsEncodedVideoChunk {
+  constructor(WebCodecsEncodedVideoChunkInit init);
+  readonly attribute WebCodecsEncodedVideoChunkType type;
+  readonly attribute long long timestamp;
+  readonly attribute unsigned long long? duration;
+  readonly attribute unsigned long byteLength;
+
+  undefined copyTo([AllowShared] BufferSource destination);
+};
+
+typedef [EnforceRange] long long WebCodecsEncodedVideoChunkInitTimestamp;
+typedef [EnforceRange] unsigned long long WebCodecsEncodedVideoChunkInitDuration;
```

Comment:
This aligns with the spec.
We want to enforce range when creating the video chunk from a dictionary.
The video chunk attribute getters should be fine since they will expose already validated values.